data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/InjectionSetpointAdderImpl.java

```diff
@@ -50,6 +51,9 @@ public InjectionSetpointAdder withSetpoint(double setpoint) {
 
     @Override
     public InjectionSetpointAdder withUnit(Unit unit) {
+        if (unit == Unit.SECTION_COUNT && setpoint < 0) {
```

Comment:
Move this to the add() method, because we're not sure that the user will set the setpoint before setting the unit. 
also check that "setpoint" is close enough to an integer (for example):
```suggestion
        if (unit == Unit.SECTION_COUNT && (setpoint < 0 || Math.abs(setpoint - Math.floor(setpoint)) > 1e-6)) {
```

Reply:
Done !